### PR TITLE
Product Creation with AI: product preview logic

### LINKED ISSUE #9810: Subscreen for showing the product preview
```[tasklist]
### Tasks
- [x] Subscreen UI
- [x] Logic to generate the product and show preview
```

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #9810 

### Description
This PR adds the logic of the preview:
1. Calls the AIRepository to start generating the product (the AI prompt is not implemented now, it'll be implemented separately)
2. Builds the property cards from the received product and updates the state.

The error handling will be done separately as well.

### Testing instructions
1. Apply the patch from below.
2. Use an eligible site for AI.
3. Open the app, then start product creation.
4. Select the AI path.
5. Click on Continue on first two screens.
6. Notice the loading, then the UI should be updated after 5 seconds with a **demo** product.

```patch
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt	(revision 813172782bed5382a21f8c93d23039398d77faf9)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt	(date 1695402763823)
@@ -53,7 +53,7 @@
                 enteredName = state.name,
                 onProductNameChanged = {},
                 onSuggestNameClicked = {},
-                onContinueClicked = {}
+                onContinueClicked = viewModel::onDoneClick
             )
         }
     }
```

### Images/gif
https://github.com/woocommerce/woocommerce-android/assets/1657201/252d00ba-ac6b-497a-958c-3062579d21b0

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt [hunk 1]
```diff
@@ -15,8 +16,14 @@ import javax.inject.Inject
 
 @HiltViewModel
 class AddProductWithAIViewModel @Inject constructor(
-    savedStateHandle: SavedStateHandle
+    savedStateHandle: SavedStateHandle,
+    aiRepository: AIRepository,
+    buildProductPreviewProperties: BuildProductPreviewProperties
 ) : ScopedViewModel(savedState = savedStateHandle) {
+    private val previewSubViewModel = ProductPreviewSubViewModel(aiRepository, buildProductPreviewProperties) {
+        saveButtonState.value = SaveButtonState.Shown
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt:24]: I noticed we show the "Save as draft" if we navigate back from preview screen. We'll probably need to add `saveButtonState.value = SaveButtonState.Hidden` inside `goToPreviousStep()`

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ai/AIRepository.kt [hunk 2]
```diff
@@ -46,7 +53,35 @@ class AIRepository @Inject constructor(
             features,
             languageISOCode
         )
-        return fetchJetpackAICompletionsForSite(site, prompt, PRODUCT_DESCRIPTION_FEATURE)
+        return fetchJetpackAICompletionsForSite(selectedSite.get(), prompt, PRODUCT_DESCRIPTION_FEATURE)
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    suspend fun generateProduct(
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ai/AIRepository.kt:60]: I think we are missing the `aiTone` here as param.

**Comment by JorgeMucientes** [comment]
Great job @hichamboushaba 🙌🏼  It works as expected! I left a couple of comments that I believe should be fixed before merging though. Although they can both be addressed in subsequent PR if you prefer. So, I won't block the PR.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by hichamboushaba** [comment]
Thanks @JorgeMucientes for the review, I'll merge this as both remarks are being handled in next PRs.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt:24]: As explained below, this is being handled in a future PR.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ai/AIRepository.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ai/AIRepository.kt:60]: As explained below, this is being handled in a future PR.

**Pull request merged**